### PR TITLE
Fix print format of rootScoped resourced in kubectl

### LINKED ISSUE #53767: Empty namespace column being shown for get clusterroles
```
$ kubectl get clusterroles --all-namespaces
NAMESPACE   NAME                                                                   AGE
            admin                                                                  57d
            cluster-admin                                                          57d
            edit                                                                   57d
            system:auth-delegator                                                  57d
            system:basic-user                                                      57d
            system:certificates.k8s.io:certificatesigningrequests:nodeclient       57d
            system:certificates.k8s.io:certificatesigningrequests:selfnodeclient   57d
            system:certificates.k8s.io:certificatesigningrequests:selfnodeserver   57d
            system:controller:attachdetach-controller                              57d
            system:controller:certificate-controller                               57d
            system:controller:cronjob-controller                                   57d
            system:controller:daemon-set-controller                                57d
            system:controller:deployment-controller                                57d
            system:controller:disruption-controller                                57d
            system:controller:endpoint-controller                                  57d
```

This is wrong, cluster scoped resources should never have a NAMESPACE column.  We need to fix and add a test to ensure this doesn't regress.

@kubernetes/sig-cli-bugs

### PR DESCRIPTION
**Pull request by lichuqiang**
**What this PR does / why we need it**:
remove _NAMESPACE_ filed when querying rootScoped resource with kubectl

**Which issue this PR fixes**

fixes #53767

**Special notes for your reviewer**:
/cc @smarterclayton 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by zjj2wry** [comment]
should update test `TestPrintHumanReadableWithNamespace` func in pkg/printers/internalversion/printers_test.go

**Comment by lichuqiang** [comment]
@zjj2wry Well, as I mentioned in #53767. The printer itself works fine, and would always behavior as we expected if only the _WithNamespace_ is true when constructing HumanReadablePrinter.
To cover the scene, the key point is to provide a factory that could pass  param _WithNamespace_ into printer dynamically when running command. I feel it not like behavior of unit test to some extent.

**Comment by zjj2wry** [comment]
previous version kubectl fix:
```
func printNamespace(obj *api.Namespace, options printers.PrintOptions) ([]metav1alpha1.TableRow, error) {
	if options.WithNamespace {
		return nil, fmt.Errorf("namespace is not namespaced")
	}
	row := metav1alpha1.TableRow{
		Object: runtime.RawExtension{Object: obj},
	}
	row.Cells = append(row.Cells, obj.Name, obj.Status.Phase, translateTimestamp(obj.CreationTimestamp))
	return []metav1alpha1.TableRow{row}, nil
```
have question about we should aware users this `resources is not namespaced`?

**Comment by lichuqiang** [comment]
>previous version kubectl fix:
func printNamespace(obj *api.Namespace, options printers.PrintOptions) ([]metav1alpha1.TableRow, error) {
	if options.WithNamespace {
		return nil, fmt.Errorf("namespace is not namespaced")
	}
	row := metav1alpha1.TableRow{
		Object: runtime.RawExtension{Object: obj},
	}
	row.Cells = append(row.Cells, obj.Name, obj.Status.Phase, translateTimestamp(obj.CreationTimestamp))
	return []metav1alpha1.TableRow{row}, nil
have question about we should let user aware this resources is not namespaced?

@zjj2wry It's a matter of behavior. It depends on whether we would like kubectl to ignore namespace param for rootScoped resources, or just reject the request with error.
I think the former is more friendly to users, @smarterclayton What's your opinion?

**Comment by zjj2wry** [comment]
i am alse like this behavior ~

File: pkg/kubectl/cmd/get.go [hunk 1]
```diff
@@ -244,7 +244,14 @@ func RunGet(f cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args [
 		}
 		info := infos[0]
 		mapping := info.ResourceMapping()
-		printer, err := f.PrinterForMapping(cmd, false, nil, mapping, allNamespaces)
+
+		// no need to print namespace for root-scoped resources
+		printWithNamespace := allNamespaces
+		if mapping.Scope != nil && mapping.Scope.Name() == meta.RESTScopeNameRoot {
```
Comment (smarterclayton) [inline pkg/kubectl/cmd/get.go]: mapping.Scope is not allowed to be nil so you don't have to check it here.

**Comment by CaoShuFeng** [comment]
/LGTM
Test OK in my environment. Thanks.

I think it would be better to add a cli test here:
https://github.com/kubernetes/kubernetes/blob/master/hack/make-rules/test-cmd-util.sh#L4147

**Comment by lichuqiang** [comment]
>I think it would be better to add a cli test here:
https://github.com/kubernetes/kubernetes/blob/master/hack/make-rules/test-cmd-util.sh#L4147

@CaoShuFeng That SGTM, I've added a test. could you give it a LGTM?

**Comment by CaoShuFeng** [comment]
/LGTM

Thanks. :)

**Comment by lichuqiang** [comment]
@dims @dshulyak Is anybody free to take a look at the PR?

**Comment by fabianofranz** [comment]
@juanvallejo looks good to you?

**Review by juanvallejo — APPROVED** [review]

**Comment by juanvallejo** [comment]
@fabianofranz this lgtm

**Comment by smarterclayton** [comment]
Looking over this again, the printer is not required to expose options that it doesn't support.  Passing WithNamespace to a cluster scoped resource is as meaningless as asking an object without labels to show its labels.  The printer is the one who should decide whether to show namespaces - PrinterOptions is a suggestion, not a mandate for this particular flag.

So I'd like to fix the printers, not the get.go command.

**Comment by smarterclayton** [comment]
Another example is Wide and AbsoluteTimestamps - a type that doesn't have "wide" columns can ignore the flag, and a type without timestamps can ignore the behavior to make it absolute.

**Pull request merged**